apps/OpenSpace/ext/launcher/src/profile/scriptlogdialog.cpp

```diff
@@ -0,0 +1,102 @@
+ *                                                                                       *
+ * Copyright (c) 2014-2021                                                               *
+ *                                                                                       *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this  *
+ * software and associated documentation files (the "Software"), to deal in the Software *
+ * without restriction, including without limitation the rights to use, copy, modify,    *
+ * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to    *
+ * permit persons to whom the Software is furnished to do so, subject to the following   *
+ * conditions:                                                                           *
+ *                                                                                       *
+ * The above copyright notice and this permission notice shall be included in all copies *
+ * or substantial portions of the Software.                                              *
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include "profile/scriptlogdialog.h"
+#include <ghoul/filesystem/filesystem.h>
+#include <openspace/engine/configuration.h>
+
+#include "profile/line.h"
+#include <openspace/scene/profile.h>
+#include <QDialogButtonBox>
+#include <QLabel>
+#include <QFile>
+#include <QTextStream>
+#include <QVBoxLayout>
+
+ScriptlogDialog::ScriptlogDialog(KeybindingsDialog* bindingDialog,
+                                                 QWidget* parent)
+    : QDialog(parent)
+    , _bindingDialog(bindingDialog)
+{
+    setWindowTitle("Scriptlog");
+    createWidgets();
+ 
+    QFile file(QString::fromStdString(absPath("${LOGS}/scriptLog.txt")));
+    if (file.open(QIODevice::ReadOnly | QIODevice::Text)) {
+        QTextStream in(&file);
+        while (!in.atEnd()) {
+            QString line = in.readLine();
+            //removing return from a few statments
+            //these are usually generated by gui panels
+            line.remove(QRegularExpression("^return "));
+            if (line.length() > 0) {
+                _scriptlogList->addItem(line);
+            }
+        }
+        file.close();
```

Comment:
I don't think it is necessary to close the file since it should be done in the destructor